src/ACTSDuplicateRemoval.cxx

```diff
@@ -17,21 +17,40 @@ namespace ACTSTracking {
  * @param tkr2 One of two tracks to compare
  * @return True if they share too many of the same Tracker Hits
  */
-inline bool tracks_equal(const edm4hep::Track& trk1, const edm4hep::Track& trk2) {
-	// Get an iterator for the Track hits of the first Track
+inline bool tracks_equal(const edm4hep::Track& trk1, const edm4hep::Track& trk2, MsgStream log) {
+        // Get an iterator for the Track hits of the first Track
         uint32_t hitOlap = 0;
         // Loop through each track hit and see if it overlaps with the second track
         for (size_t itrackHit = 0; itrackHit < trk1.trackerHits_size(); ++itrackHit) {
-                if (std::find(trk2.trackerHits_begin(), trk2.trackerHits_end(), trk1.getTrackerHits(itrackHit)) != trk2.trackerHits_end()) {
+                auto it = std::find_if(trk2.trackerHits_begin(), trk2.trackerHits_end(), [&](const auto& hit2){
+                        return hitEqual(trk1.getTrackerHits(itrackHit), hit2);
+                });
+                if (it != trk2.trackerHits_end()) {
                         hitOlap++; // If it does overlap, increment hitOlap
                 }
         }
 
         // Smaller track count
         uint32_t size = std::min(trk1.trackerHits_size(), trk2.trackerHits_size());
+        log<<MSG::WARNING<<hitOlap*2<<"   "<<size<<endmsg;
         return 2 * hitOlap > size;  // half of smaller track belong to larger track
 }
 
+/**
+ * @brief Workaround function to get around comparision issue
+ * @TODO: This is to get around an issue in edm4hep
+ * @param hit1 A hit
+ * @param hits A hit
+ */
+bool hitEqual(const edm4hep::TrackerHit& hit1, const edm4hep::TrackerHit& hit2) {
+        return          hit1.getCellID() == hit2.getCellID() &&
+                        hit1.getType() == hit2.getType() &&
+                        hit1.getQuality() == hit2.getQuality() &&
+                        hit1.getTime() == hit2.getTime() &&
+                        hit1.getEDep() == hit2.getEDep() &&
+                        hit1.getEDepError() == hit2.getEDepError() &&
+                        hit1.getPosition() == hit2.getPosition();
+}
```

Comment:
In principle `TrackerHit` has an `operator==`. However, that will only return true if the two tracker hits are truly the same, i.e. you only have two handles that point to the same object. The fact that you have to implement this, suggests, that at some point tracker hits are cloned(?), which should in principle not be necessary, unless I am missing something here.

Reply:
Yes! I saw the operator== but it never seemed to work. I assumed it was a bug in edm4hep and that it would be fixed in the new release, but thinking about it some more, it probably is coming from 1 of 2 places:
1. The merging of collections creates new objects right now, which makes the operator== not work. So switching to subsetCollections should fix that.
2. The issue might be coming from lcio2edm4hep converter, though. If this is the case, I am unsure of how to fix it other than to leave this.

Reply:
There's also the question of as I move through sections like the Deduplicator and Filter, I want to only keep certain tracks in the output collections. At first I tried just adding the tracks I wanted to the new output collections, but I got an error saying 
`ERROR Object already in a collection. Cannot add it to a second collection`
So now I have it making new tracks from the same data and adding those. That doesn't feel like the right way to do things, especially if I want to keep it as the same object. Thoughts?

Reply:
That would exactly be the use case for a subset collection. In EDM4hep (more precisely for any podio generated datamodel), there is a builtin assumption that each element exists exactly once. That is because it is uniquely identified by two things: 1) which collection it belongs to (`collectionID`) and 2) where in that collection it is located (`index`). A subset collection is a way to group elements of a given type in a different way, because that doesn't assign a new `collectionID` nor a new `index`.

This works quite well as long as we don't have to change any data of the objects, e.g. the tracks. If I understand the filtering and deduplication correctly, you don't have to do that, right? You always either keep the full track or discard the full track without having to change any of the data that is stored in that track. If that is the case a subset collection should work and do what you would expect it to do.

Things get a bit more tricky once you get to a stage where you also have to change some data of the original tracks. That is not possible in EDM4hep, because it makes multi threaded execution a nightmare. In that case we would be back to needing a new collection and putting some `clone`d tracks into them. We can change the clones again since we are certain that no other thread can see them before the end of the algorithm.

Currently there is no really nice way to do that, other than manually going through the collection and `clone` the tracks that you need. But we can certainly put some utilities for facilitating that into podio, especially if there is a concrete use case that we want to tackle.

Reply:
This is perfect! Thank you. I'll try implementing some subset collection around where I was using cloning before